src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -3904,21 +3939,36 @@ export class ResultsViewPageStore {
         },
     });
 
-    readonly treatmentLinkMap = remoteData<{ [treatmentId: string]: string }>({
-        await: () => [this.molecularProfilesInStudies],
+    readonly genericAssayEntitiesGroupByGenericAssayTypeLinkMap = remoteData<{
+        [genericAssayType: string]: { [stableId: string]: string };
+    }>({
+        await: () => [this.genericAssayEntitiesGroupByGenericAssayType],
         invoke: async () => {
+            const linkMapGroupByGenericAssayType: {
+                [genericAssayType: string]: { [stableId: string]: string };
+            } = {};
             if (
-                this.molecularProfilesInStudies.result &&
-                this.molecularProfilesInStudies.result.length > 0
+                !_.isEmpty(
+                    this.genericAssayEntitiesGroupByGenericAssayType.result
+                )
             ) {
-                const treatments = await fetchTreatmentByMolecularProfileIds(
-                    this.molecularProfilesInStudies.result
+                const genericAssayTypes = _.keys(
+                    this.genericAssayEntitiesGroupByGenericAssayType.result
                 );
-                const linkMap: { [treatmentId: string]: string } = {};
-                treatments.forEach(({ treatmentId, refLink }) => {
-                    linkMap[treatmentId] = refLink;
+                genericAssayTypes.map(genericAssayType => {
+                    const linkMap: { [stableId: string]: string } = {};
+                    this.genericAssayEntitiesGroupByGenericAssayType.result![
+                        genericAssayType
+                    ].forEach(entity => {
+                        // if entity meta contains reference url, add the link into map
+                        linkMap[entity.stableId] =
+                            'URL' in entity.genericEntityMetaProperties
```

Comment:
entity.genericEntityMetaProperties.URL || ""

Reply:
We don't have a predefined type for `genericEntityMetaProperties`, `genericEntityMetaProperties` is an object returned from the backend API, we don't know what's in it before we try to get the data. So I may not do as you suggested.